src/main/java/com/wyden/findyourhome/entities/Customer.java

```diff
@@ -0,0 +1,130 @@
+package com.wyden.findyourhome.entities;
+
+import jakarta.persistence.*;
+import java.util.List;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Entity
+@Table(name = "Custumer")
+public class Customer implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Integer id;
+
+    private String name;
+    private String email;
+    private String phone;
+
+    private String cpf;
+    private String cnpj;
+
+    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
+    private List<Telephone> telephones;
+
+    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
+    private List<Advertisement> advertisements;
+
+    public Customer(
+          Integer id, 
+          String name, 
+          String email, 
+          String phone, 
+          String cpf, 
+          String cnpj, 
+          List<Telephone> telephones, 
+          List<Advertisement> advertisements
+     ){
+          this.id = id;
+          this.name = name;
+          this.email = email;
+          this.phone = phone;
+          this.cpf = cpf;
+          this.telephones = telephones;
+          this.advertisements = advertisements;
+     }
+
+    public Customer() {
+    }
+
+    public int getId() {
+        return this.id;
+    }
+
+    public void setId(int id) {
```

Comment:
Só remover o setId por causa do banco de dados.
Vamos evitar permitir alteração de chave primária no banco? O que acontece se chave for alterada para outra que já existe? Quais as consequencias se a chave alterada for a chave estrangeira de outra tabela?

Reply:
É o costume de pedir a extensão do vscode para criar todos os getters e setters, alguns passam...t.t perdon hauhauha

Reply:
Removido